### DISCUSSION TITLE
How to control cache in Nextjs app router and trpcProxyClient?

### DISCUSSION THREAD

**Discussion by UROjQ6r80p** [trpc/trpc#5108] [Q&A]
How can I control cache behaviour with trpc proxy client in nextjs 13 app router?

```
import type { AppRouter } from '...'
import { createTRPCProxyClient, httpBatchLink } from "@trpc/client";

export const client = createTRPCProxyClient<AppRouter>({
    links: [
      httpBatchLink({
        url: 'http://localhost:8080/trpc',
      }),
    ],
});

await fetch('http://localhost:3000/api/something, { cache: 'no-cache' }) // always returns new response

await client.getPosts.query() // returns stale data
```

**Comment by jeff-computers** [comment]
Bump, there are no good docs or answers to this.

**Comment by KenSoftDev** [comment]
https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#fetching-data-on-the-server-with-third-party-libraries

>In cases where you're using a third-party library that doesn't support or expose fetch (for example, a database, CMS, or ORM client), you can configure the caching and revalidating behavior of those requests using the [Route Segment Config Option](https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config) and React's cache function.

I think it would be beneficial to use 'Route Segment Config' and React Cache.

- https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config
- https://react.dev/reference/react/cache
